### PR TITLE
Use fold instead of mutable state

### PR DESCRIPTION
**Pull request by crisptrutski**
Apply a slightly more functional approach to accumulating the the table and column names.

### CODE REVIEW

File: java/com/metabase/macaw/AstWalker.java [hunk 1]
```diff
@@ -207,15 +204,17 @@
  * the conventional visitor pattern, instead providing the `callbacks` map This lets Clojure code use a normal Clojure
  * map and functions to implement the necessary behavior; no `reify` necesary.
  */
-public class ASTWalker implements SelectVisitor, FromItemVisitor, ExpressionVisitor,
+public class AstWalker<Acc> implements SelectVisitor, FromItemVisitor, ExpressionVisitor,
```
Comment (crisptrutski) [inline java/com/metabase/macaw/AstWalker.java:207]: This is the correct way to Capitalise a Tla 😛 

Appeal to authority based for languages with Java in the name:

- https://www.oreilly.com/library/view/java-8-pocket/9781491901083/ch01.html
- https://google.github.io/styleguide/jsguide.html#naming-camel-case-defined

Rejection of authority for languages with a lesser Clojure interop story.

- https://learn.microsoft.com/en-us/previous-versions/dotnet/netframework-1.1/141e06ef(v=vs.71)

File: java/com/metabase/macaw/AstWalker.java [hunk 2]
```diff
@@ -245,15 +245,17 @@ public ASTWalker(Map<Keyword, IFn> callbacksWithKeywordKeys) {
     public void invokeCallback(String callbackName, Object visitedItem) {
         IFn callback = this.callbacks.get(callbackName);
         if (callback != null) {
-            callback.invoke(visitedItem);
+            //noinspection unchecked
+            acc = (Acc) callback.invoke(acc, visitedItem);
```
Comment (crisptrutski) [inline java/com/metabase/macaw/AstWalker.java:249]: This is perhaps a bit too reckless, there's nothing stopping Clojure code from changing the type, perhaps in subtle ways like `clojure.lang.PersistentArrayMap` to `clojure.lang.PersistentHashMap`. Good enough for now though.

File: src/macaw/core.clj [hunk 3]
```diff
@@ -12,20 +12,20 @@
 
 (set! *warn-on-reflection* true)
 
+(defn- walk-query [parsed-query callbacks val]
```
Comment (crisptrutski) [inline src/macaw/core.clj:15]: Sugar is essential 🍬

File: java/com/metabase/macaw/AstWalker.java (hunk 1 above)
Comment (crisptrutski) [reply java/com/metabase/macaw/AstWalker.java:207]: Oh, I see you inherited it from JSqlParser. Probably not worth being inconsistent then as power users of the library might work with a combination of our classes and those we're importing.

**Review by tsmacdonald — APPROVED** [review]
:sunglasses:

**Pull request merged**